### PR TITLE
Parsing domain username

### PR DESCRIPTION
**Pull request by NimishaS**
w.r.t. https://github.com/chef/knife-azure/issues/254

### CODE REVIEW

File: lib/chef/knife/azure_server_create.rb [hunk 1]
```diff
@@ -957,9 +955,7 @@ def create_server_def
             server_def[:azure_domain_name] = $2
             server_def[:azure_domain_user] = $1
           else
-            # Format error.
-            ui.error("Format error for --azure-domain-user option. Supported format are user principal name (UPN) format (user@fully-qualified-DNS-domain) or the fully-qualified-DNS-domain\\username format")
-            exit 1
+            server_def[:azure_domain_user] = locate_config_value(:azure_domain_user)
```
Comment (adamedx) [inline lib/chef/knife/azure_server_create.rb]: Is our way of constructing the username that we pass to Azure going to allow us to specify a user that's not in the domain we're trying to join? Users from one domain can be authorized to join systems to another domain if there is a trust between those domains.

**Comment by adamedx** [comment]
So I think we should accept two name formats for user name:

* `user`: this implies `user@domain.youwantojoin.com`
* `user@mydomain.mycompany.com`: this is the User Principal Format (UPN) -- enabling this allows you to join the domain using credentials for a user from another domain than the one you are trying to join, which is a real scenario.

Also, we should forbid `domain\user` format -- it is ambiguous, and people only use it because they can use name services like `WINS` within an on-premises network -- that name format generally doesn't work in Internet or Azure cloud scenarios. DNS names could be used instead, but people rarely do that, and UPN provides the same capability and is also used elsewhere in Azure UX. Our cli help and docs should be very clear about what name formats are allows for these parameters and which aren't.

**Comment by adamedx** [comment]
@vinyar, see comment above about name formats that we will allow -- your users should specify `user@mydomain.mycompany.com` instead of `domain\user` when specifying the username, or just specify `user` without a domain if that user is in the domain you are trying to join (either format will work if the domains are the same).

**Comment by vinyar** [comment]
Could you include readme.md and internal `-h` help output updates into the commit?

Thanks :+1:

**Comment by adamedx** [comment]
Seems like a good idea @vinyar. @NimishaS, could you add a domain join example to the readme that uses upn format? The CLI help should also be very clear about the expected format as well.

**Comment by adamedx** [comment]
Looks good once the tests are passing again.

**Pull request merged**